Easy deployment of MPCAutofill using Docker

fklemme opened the pull request:
This pull-request adds Docker and Docker-Compose files for an easy setup and deployment of MPCAutofill. Besides the configuration of the `drives.csv` and the Google Service Account, most other steps are automated. Please have a look at the Readme for further information.

The following goals are aimed to be achieved in this PR (unchecked are to-do's):
- [x] Minimal software installation, requires only docker and docker-compose.
- [x] Easy setup, only individual configurations are required, such as setting up your `drives.csv`.
- [x] Decoupled, introduces only minimal changes to the original code base. All additions can be ignored if Docker is not used.
- [x] Persistent databases, enabling fast pause and resume of containers.
- [x] Chronjobs, keeping the containers up to date automatically.
- [x] Updated Readme, explaining the installation using Docker.

Feel free to test the Docker setup and propose changes. Any help or feedback is welcome.

bwsinger commented:
> Updated Readme, hinting towards the possibility of using Docker.

Once my README PR is accepted, I'll make further edits to it and the docker README to make users aware of their options.

ndepaola commented:
holy shit i love you - can't wait to test this out tomorrow!

fklemme commented:
> Update client (Google Scripts) as well? Is this needed?

I tested the current state with the already available client release on this Github page and it worked just fine. Apparently, there is no interdependency between the server-side Google Service Account and the called Google Scripts in the client. Therefore, this point does not require any handling here and is removed from the TODOs.

fklemme commented:
I think the PR is ready to be merged. Let me know if further changes are required.

@bwsinger, I added a small note to the README but I think it should be easy for you to merge the changes back into your PR and change them to your needs.

ndepaola commented on README.md:
```diff
@@ -28,3 +28,6 @@ Other:
 * `0 0 * * SUN bash /root/mpc-autofill/sync_dfcs`
 8. Deploy two Google Script according to the code specified in `autofill.py` and adjust the URLs in that script to point to your GS endpoints
+
+## Using Docker
```
let's move this to the top of the readme since i imagine most people will use docker to run the project

fklemme replied:
I completely restructured the main README and merged the one from the docker sub-folder. Now, the first part is about preparative steps (setting up Google Server Account, etc.) and the second part is about the technical installation. Have a look at the rendered README at https://github.com/fklemme/mpc-autofill. I left the manual installation part (which is now at the end) to PR #33.

ndepaola replied:
looks good! let's add a note for windows installation though. PR #33 might want to rebase onto this branch?

bwsinger replied:
Yeah, I think that's the plan. Let's get this PR merged. Then I'll use bits and pieces from my verbose README to flesh out this one.

ndepaola commented:
![image](https://user-images.githubusercontent.com/3079166/134804148-688ce868-1ebd-4756-8ce3-c67aec760f34.png)

wondering if you've seen this one before?

fklemme commented:
> let's add a note for windows installation though.

> wondering if you've seen this one before?

Gotta admit, I didn't even know that docker/docker-compose exists for Windows. :P Let me try to install it and see if I can reproduce the error.

fklemme commented:
To my very surprise, running docker-compose under Windows worked absolutely flawlessly. :D I added some instructions to the README.

Apparently, the scripts `docker/django/check_*.sh` are not found in your setup. Can you check if you have them in your repo? I added them just recently, so maybe something went wrong when you updated your repo?

On a clean checkout, the scripts should actually stop docker and yield a human-readable error message like this:

![docker_unconfigured](https://user-images.githubusercontent.com/5053254/134813227-2adcc429-08fa-42e6-b768-77bb441dea42.png)

But after installing `client_secrets.json` and `drives.csv`, docker should run just fine.

fklemme commented:
@ndepaola: What does it mean for MPCAutofill if a Google Drive has `public = FALSE` in the `drives.csv`? When would I use this option?

fklemme commented:
To easy. :sunglasses:

![docker_demo_windows](https://user-images.githubusercontent.com/5053254/134819339-56d89915-1baa-4a28-9710-6bdd6b13eb34.gif)

ndepaola commented:
weird that it didn't work when i checked out your branch but did work when i downloaded your repo as .zip and went from there! did i say i love you btw? this works beautifully and i can't wait for other people to try it!!

ndepaola commented:
> 
> 
> @ndepaola: What does it mean for MPCAutofill if a Google Drive has `public = FALSE` in the `drives.csv`? When would I use this option?

`public=FALSE` means the drive's link won't be visible on the contributions page or search settings popup

fklemme commented:
> weird that it didn't work when i checked out your branch but did work when i downloaded your repo as .zip and went from there! did i say i love you btw? this works beautifully and i can't wait for other people to try it!!

Love you for building MPCAutofill! <3 :smile:
